### PR TITLE
Enable Netty 4 extensions

### PR DESCRIPTION
**Pull request by jasontedor**
This pull request enables some extensions to the Netty 4 implementation.

Relates #19526

### CODE REVIEW

File: modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4Transport.java [hunk 1]
```diff
@@ -326,6 +311,14 @@ protected void initChannel(SocketChannel ch) throws Exception {
         serverBootstraps.put(name, serverBootstrap);
     }
 
+    protected ChannelHandler getServerChannelInitializer(String name, Settings settings) {
```
Comment (abeyad) [inline modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4Transport.java]: is there a reason not to inline this method and the one below?

Comment (jasontedor) [inline modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4Transport.java]: Yes, to enable extensions.

Comment (abeyad) [inline modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4Transport.java]: Discussed this with @jasontedor, he explained that this class can be extended and these methods need to be overridable.

**Comment by abeyad** [comment]
LGTM

File: modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4Transport.java (hunk 1 above)
Comment (javanna) [inline modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4Transport.java]: shall we document the fact that these methods can be extended?

**Pull request merged**